lib/after_commit_action.rb

```diff
@@ -17,6 +17,11 @@ module AfterCommitAction
   end
 
   def execute_after_commit(&block)
+    if ActiveRecord::Base.connection.open_transactions == 0 ||
+      ActiveRecord::Base.connection.add_transaction_record(self).nil?
+      return block.call
+    end
```

Comment:
Ok, that line of code is what I needed to see. :smile: 

What you are saying is correct, however that is intentional -- if there is a transaction but the current record is not in it, we want to simply add it so that the `after_commit` hook runs after the commit. Given the intention behind this gem, that seems like the more correct thing to do. Also, your test is still green for it.

See #5 for what I mean.